src/Feature/Layout/Feature_Layout.re

```diff
@@ -0,0 +1,284 @@
+open Utility;
+
+type direction =
+  | Up
+  | Left
+  | Down
+  | Right;
+
+type t('content) =
+  | Split([ | `Horizontal | `Vertical], list(t('content)))
```

Comment:
I wonder if we will need to add a `weight` to the Split as well? For example:

```
| A | B |
--------
| C | D |
```
(ie, `Split(Horizontal, [`Vertical([splitA, splitB]), `Vertical([splitC, splitD])])`)

- if we wanted to resize the _height_ of A/B or C/D - it seems like that would need to live on the `Split`. Just something to think about as we get into resizing

Reply:
Yeah, that's a good point. The weight should probably be associated with the "child", not the window, so the list of children would be more like `list((float, t('content)))`. I'll leave it as-is as it's not actually used yet, and will fix it when I do the resizing shortly.